catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RegistryPublicationServiceImpl.java

```diff
@@ -142,7 +142,11 @@ public void update(Metacard metacard) throws FederationAdminException {
                 .collect(Collectors.toCollection(HashSet::new));
 
         if (CollectionUtils.isNotEmpty(locations)) {
-            federationAdminService.updateRegistryEntry(metacard, locations);
+            try {
+                federationAdminService.updateRegistryEntry(metacard, locations);
+            } catch (FederationAdminException e) {
+                federationAdminService.addRegistryEntry(metacard, locations);
```

Comment:
The update would only be throwing an exception after a remote call too, right? If the odds are vanishingly small then I'd be okay leaving it this way (with a comment explaining the flow and the low likelihood of a failure); otherwise, it would be better to switch to an explicit call to check.